plugins/libplugin-pay.c

```diff
@@ -2892,13 +2908,17 @@ static bool routehint_excluded(struct payment *p,
 		 * know the exact capacity we need to send via this
 		 * channel, which is greater than the destination.
 		 */
-		for (size_t j = 0; j < tal_count(hints); j++) {
-			if (!short_channel_id_eq(hints[j].scid.scid, r->short_channel_id))
+		struct channel_hint *hint;
+		struct channel_hint_map_iter iter;
+		for (hint = channel_hint_map_first(hints, &iter);
```

Comment:
Why loop over the hash map here? Couldn't we use lookup by hash?

Reply:
The channel_hint_map is indexed by scid and direction, while here we only have the direction. An alternative could be to make a channel hint map based on scid only, with a struct with 2 fields for the different directions. Then we wouldn't have to iterate here. 

This part of the code is used once per path though, and not with every step of the dijkstra procedure. The things fixed in this PR were called approximately 35000 times per payment, while this one is called once per payment. So the performance overhead is a lot less.

Reply:
Actually it seems like we should be able to get the hint by getting from the channel hints map with 0 or 1 direction. I'll dig which one to use here. It seems weird that the direction wasn't used before, you could have gotten the route hint in the wrong direction and assumed the routehint was excluded 🤔

Reply:
Once per payment is not bad, I have no issues with leaving the loop here.

But the direction of the hint (the `r`) is already defined by the ordering of the nodes on both ends of the channel.
I think the previous code was lazy to check only the short channel id, maybe considering that we usually get information
about channels in one direction and rarely we get the other direction. IDK

Reply:
@Lagrang3 I added a commit that attempts to get the route hint from the map directly. Can you check that out? Especially the pubkeys used for the comparison there.